### PR TITLE
Added Firefox Adapter

### PR DESCRIPTION
**Pull request by hitesh96db**
I've made some changes in the interface so as to adapt to Jetpack. I've also included functionality in the Firefox Adapter.

### CODE REVIEW

File: shared/javascripts/context_messenger.js [hunk 1]
```diff
@@ -275,10 +284,10 @@ if (Privly === undefined) {
   };
 
   /** @inheritdoc */
-  ChromeAdapter.prototype.setListener = function (callback) {
+  ChromeAdapter.prototype.setListener = function (messenger) {
     chrome.runtime.onMessage.addListener(function (payload) {
       // for Chrome, we needn't check origin since it is always from a trust origin
-      callback(payload);
+      messageHandler(payload);
```
Comment (SummerWish) [inline shared/javascripts/context_messenger.js]: When `messenger` would be called? is it `messageHandler(payload, messenger)`?

Comment (hitesh96db) [inline shared/javascripts/context_messenger.js]: Actually, Chrome and Safari don't need the messenger option, so it's okay if they don't call `messageHandler` with the messenger.

Comment (SummerWish) [inline shared/javascripts/context_messenger.js]: Emm. I didn't express my questions well. I mean here, `messenger` is a callback, which will be called when a message is received..  I quickly looked at the code but cannot find where it is passed to.. so I guess it won't be called?

Comment (hitesh96db) [inline shared/javascripts/context_messenger.js]: Oh, `messenger` is not a callback. the `setListener` no longer receives a callback. It instead receives an API object that allows it to listen to a message.

Comment (SummerWish) [inline shared/javascripts/context_messenger.js]: Is this API object consistent across platform? If not, I'm afraid we have to fill the gap. The purpose of this library is to provide a *consistent* message interface for all platforms. I mean, the code related to sending and receiving messages shouldn't need to change when platform changes.

Comment (SummerWish) [inline shared/javascripts/context_messenger.js]: Could you give a documentation link about sending and receiving messages on JetPack platform? Let's work together to find a good solution.

Comment (hitesh96db) [inline shared/javascripts/context_messenger.js]: Can you come on IRC ?

Comment (SummerWish) [inline shared/javascripts/context_messenger.js]: Online

**Comment by hitesh96db** [comment]
@SummerWish I tried out what we discussed yesterday and it works(haven't tried sendResponse yet) , but I don't like the idea of mixing up the two(Jetpack & context messenger) messaging interfaces, which makes it confusing to write.
Instead, how about making the worker listen everytime I push it into the array of workers ? so when a worker is available,  I do this - 
<pre>
// Page-Mod
onAttach: function(worker) {
  worker.on("detach", popFunction());
  Privly.message.currentAdapter.pushWorker(worker);
};
// Push Worker
FirefoxAdapter.prototype.pushWorker = function(worker) {
  this.workers.push(worker);
  worker.port.on("privly", function(message) {
    FirefoxAdapter.prototype.listener(message);
  });
}
// Sending Message
Use all the workers to send the message.
// Receiving Message
Call all the user specified listeners.
</pre>

**Comment by SummerWish** [comment]
@hitesh96db It seems great :-)

**Comment by hitesh96db** [comment]
@SummerWish I've updated the PR. It seems to be working correctly :)

**Comment by SummerWish** [comment]
Congrats :-)

**Pull request merged**